docs/streams/subscriptions.md

```diff
@@ -34,14 +34,43 @@ There are two important things to be aware of when using shared mode:
 
 ![stream-shared-subscriptions](/img/stream-shared-subscriptions.png)
 
-### Failover
+## Failover
 
 In _failover_ mode, multiple consumers can attach to the same subscription. A master consumer is picked for the stream to receives messages. When the master consumer disconnects, all non-acknowledged and subsequent messages are delivered to the next consumer in line.
 
 In the diagram below, Consumer-C-1 is the master consumer while Consumer-C-2 would be the next in line to receive messages if Consumer-C-1 disconnected.
 
 ![stream-failover-subscriptions](/img/stream-failover-subscriptions.png)
 
+## Key_Shared
+
+In Key_Shared type, multiple consumers can attach to the same subscription. Messages are delivered in a distribution across consumers and message with same key or same ordering key are delivered to only one consumer. No matter how many times the message is re-delivered, it is delivered to the same consumer. When a consumer connected or disconnected will cause served consumer change for some key of message.
+
+Note that when the consumers are using the Key_Shared subscription type, you need to disable batching or use key-based batching for the producers. There are two reasons why the key-based batching is necessary for Key_Shared subscription type:
+
+-The broker dispatches messages according to the keys of the messages, but the default batching approach might fail to pack the messages with the same key to the same batch.
+-Since it is the consumers instead of the broker who dispatch the messages from the batches, the key of the first message in one batch is considered as the key of all messages in this batch, thereby leading to context errors.
+
+The key-based batching aims at resolving the above-mentioned issues. This batching method ensures that the producers pack the messages with the same key to the same batch. The messages without a key are packed into one batch and this batch has no key. When the broker dispatches messages from this batch, it uses NON_KEY as the key. In addition, each consumer is associated with only one key and should receive only one message batch for the connected key. By default, you can limit batching by configuring the number of messages that producers are allowed to send.
```

Comment:
Is NON_KEY a constant or something that should be in code format? Where would you set it up?

Reply:
Pulsar would handle it internally.

Reply:
If it is handled internally, then do we need to tell the users about it?

Reply:
@chandrashekhar-macrometa @diana-macrometa if it's something that we handle internally i.e. transparent to the user, than we should not add it to the documentation